### ISSUE TITLE
Occasional failures with Python 2, triggered by cleanup of imported modules in verify_imports

### ISSUE THREAD

**Issue by geimer** [easybuilders/easybuild-framework#3542]
When using EasyBuild with Python 2, I'm experiencing occasional failures with a traceback like this:
```
[...]
  File ".../easybuild/framework/easyconfig/easyconfig.py", line 2537, in det_full_module_name
    require_result=require_result)
  File ".../easybuild/framework/easyconfig/easyconfig.py", line 2509, in _det_module_name_with
    mod_name = mns_method(ec)
  File ".../easybuild/tools/module_naming_scheme/hierarchical_mns.py", line 89, in det_full_module_name
    return os.path.join(self.det_module_subdir(ec), self.det_short_module_name(ec))
AttributeError: 'NoneType' object has no attribute 'path'
```
The exact circumstances still seem to be unclear, but according to @Flamefire it may be related to https://stackoverflow.com/questions/29107470/imported-modules-becomes-none-when-replacing-current-module-in-sys-modules-using

**Comment by Flamefire** [comment]
In short: Unloading modules is not supported in Python: https://bugs.python.org/issue9072
More general: `sys.modules` mustn't be modified manually. But we do that in https://github.com/easybuilders/easybuild-framework/blob/b7031917445a536e591f593d1661168bd8cc495c/easybuild/tools/include.py#L146-L147

This is fatal in Python < 3.4 which clears the modules dict on unload: https://stackoverflow.com/questions/29107470/imported-modules-becomes-none-when-replacing-current-module-in-sys-modules-using and https://stackoverflow.com/questions/29107470/imported-modules-becomes-none-when-replacing-current-module-in-sys-modules-using

In Python 3.4+ the unloading seems to happen differently (or maybe even work at all). But I don't see the reasoning here to unload them, it just leads to more work loading them later again

Solutions:

- Don't modify `sys.modules`, i.e. keep the checked modules loaded
- Skip the check (do nothing in `verify_imports`) for Python < 3.4
- Remove that whole function

**Comment by boegel** [comment]
@geimer I think you mentioned that this issue is new, and it only started occurring recently. Can you provide some more info on that?
With which EasyBuild version did this start popping up?

**Comment by geimer** [comment]
@boegel With my configuration, the issue started to appear with https://github.com/easybuilders/easybuild-framework/commit/ff3d33f4c592d01655f992a09c54260c4b74201d. However, if this is really related to GC, bisecting may lead to a totally different commit if, for example, your custom easyblocks import different modules than mine. GC is more or less uncontrollable and kicks in whenever it likes to...

**Comment by boegel** [comment]
@geimer Fix was merged in #3544, please check if you're still seeing the issue with current `develop`.

**Comment by geimer** [comment]
@boegel With the fix applied, I am no longer able to reproduce this. However, since GC activity depends on the phase of the moon as well as the sum of digits of the outside temperature and humidity, it's impossible to say whether the issue is gone once and forever. But let's cross fingers and hope for the best :crossed_fingers: :wink:
